insonmnia/locator/storage/inmemory/storage.go

```diff
@@ -0,0 +1,97 @@
+package inmemory
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/zap"
+
+	ds "github.com/sonm-io/core/insonmnia/locator/datastruct"
+)
+
+var errNodeNotFound = errors.New("a Node with the given Eth address is not found")
+
+type Storage struct {
+	cleanupPeriod time.Duration
+	nodeTTL       time.Duration
+
+	logger *zap.Logger
+
+	mx sync.RWMutex
+	db map[common.Address]*ds.Node
+}
+
+func NewStorage(cleanUpPeriod, nodeTTL time.Duration, logger *zap.Logger) *Storage {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+
+	s := &Storage{
+		cleanupPeriod: cleanUpPeriod,
+		nodeTTL:       nodeTTL,
+
+		logger: logger,
+		db:     make(map[common.Address]*ds.Node),
+	}
+	go s.cleanExpiredNodes()
+
+	return s
+}
+
+func (s *Storage) Put(n *ds.Node) {
+	s.mx.Lock()
+	defer s.mx.Unlock()
+
+	n.TS = time.Now()
+	s.db[n.EthAddr] = n
```

Comment:
sdbnethaddrn

Reply:
It might have felt a joke, but the code has lived there since the where begging, all I did I just moved to another location :) We first need to decide whether we need inmemory storage at all before thinking about changing it.